src/main/java/com/google/cloud/genomics/utils/Contig.java

```diff
@@ -148,5 +184,78 @@ public Contig apply(String contigString) {
+   * @param variantSetId - the id of the variant set for which to create shards
+   * @param sexChromosomeFilter - whether or not to include the sex chromosomes in the list of shards
+   * @return a shuffled list of shards
+   * @throws IOException
+   */
+  public static List<Contig> getAllShardsInVariantSet(final Genomics genomics, final String variantSetId,
+      final SexChromosomeFilter sexChromosomeFilter) throws IOException {
+    return getAllShardsInVariantSet(genomics, variantSetId, sexChromosomeFilter, DEFAULT_NUMBER_OF_BASES_PER_SHARD); 
+  }
+  
+  /**
+   * Get a list of Contigs representing sharded windows of all genomic regions within the variant set.
+   * 
+   * These are useful for pipelines that operate in parallel against data within a specific variant set.
+   * 
+   * @param genomics - the genomics client
+   * @param variantSetId - the id of the variant set for which to create shards
+   * @param sexChromosomeFilter - whether or not to include the sex chromosomes in the list of shards
+   * @param numberOfBasesPerShard - the maximum size of the shard window in terms of number of bases
+   * @return a shuffled list of shards
+   * @throws IOException
+   */
+  public static List<Contig> getAllShardsInVariantSet(final Genomics genomics, final String variantSetId,
+      final SexChromosomeFilter sexChromosomeFilter, final long numberOfBasesPerShard) throws IOException {
+    List<Contig> contigs = getContigsInVariantSet(genomics, variantSetId, sexChromosomeFilter);
+    return getAllShardsForContigs(contigs, numberOfBasesPerShard);
+  }
+  
+  /**
+   * Get a list of Contigs representing sharded windows of the specified genomic regions expressed in the command line argument.
+   * 
+   * These are useful for pipelines that operate in parallel against data within a specific variant set or read group set.
+   * 
+   * @param contigsArgument - the command line parameter expressing the specified genomic regions
+   *                            format is chromosome:start:end[,chromosome:start:end]
+   * @return a shuffled list of shards
+   */
+  public static List<Contig> getSpecifiedShards(final String contigsArgument) {
+    return getSpecifiedShards(contigsArgument, DEFAULT_NUMBER_OF_BASES_PER_SHARD);
+  }
+
+  /**
+   * Get a list of Contigs representing sharded windows of the specified genomic regions expressed in the command line argument.
+   * 
+   * These are useful for pipelines that operate in parallel against data within a specific variant set or read group set.
+   * 
+   * @param contigsArgument - the command line parameter expressing the specified genomic regions
+   *                            format is chromosome:start:end[,chromosome:start:end]
+   * @param numberOfBasesPerShard - the maximum size of the shard window in terms of number of bases
+   * @return a shuffled list of shards
+   */
+  public static List<Contig> getSpecifiedShards(final String contigsArgument, final long numberOfBasesPerShard) {
```

Comment:
Agreed on using fewer synonyms.  There are times though that we have both contigs and shards.  Take a look at the example below and let me know what you think?

* https://github.com/deflaux/dataflow-java/blob/refactor-part1/src/main/java/com/google/cloud/genomics/dataflow/pipelines/CountReads.java#L182
* https://github.com/deflaux/dataflow-java/blob/refactor-part1/src/main/java/com/google/cloud/genomics/dataflow/pipelines/CountReads.java#L195 the sharding happens in a different class